### DISCUSSION TITLE
Additional RPi Pico W OTA Example: OTALedFlash

### DISCUSSION THREAD

**Discussion by djaus2** [earlephilhower/arduino-pico#2260] [General]
In GitHub [djaus2/Soft-ata/Misc](https://github.com/djaus2/Soft-ata/tree/master/Misc/OTALedFlash)  

An example Sketch for OTA (Over-The-Air) updating of a RPi Pico W.
Similar to examples in /libraries/ArduinoOTA/examples especially the first 2. This one just uses the Inbuilt LED with coded flashes to indicate where the sketch is at. Number of pulses and length of each is different at different points in boot/OTA events.

**Comment by earlephilhower** [comment]
Maybe we could output Morse code instead! :laughing:

Anyway, this does freeze the OTA process while flashing.  Look at the `Ticker` library for a way of doing that timed flashing w/o needing to busywait the device.  I'm a little worried that some OTA processes may think the device is DOA if it hangs w/o response for too long...

**Comment by djaus2** [comment]
Yeah I thought of Morse Code!
I did find if the OTA start event handler flashing took too long there was an upload error, so I made it short.

David Jones (MVP)
________________________________

**Comment by djaus2** [comment]
And I'll have a look at the Ticker library.
What happens if a process is running in the second CPU when an OTA update occurs?

David Jones (MVP)
________________________________

**Reply by earlephilhower** [reply]
Absolutely nothing, the 2nd core isn't aware.  OTA just writes a file to the internal LittleFS filesystem.  When all data's in the file, it then reboots.  This ensures all-or-nothing updates.

**Comment by djaus2** [comment]
Ok thanks, that is quite clear.

David Jones (MVP)
________________________________
